src/socket_peer.c

```diff
@@ -24,18 +24,15 @@
  * SOFTWARE.
  */
 
-#include <arpa/inet.h>
-#include <stdint.h>
-#include <unistd.h>
+#include "socket_peer.h"
 
-#include "alloc.h"
-#include "buffered_reader.h"
 #include "compiler.h"
-#include "jet_endian.h"
+#include INET
```

Comment:
There is already a file jet_endian.h providing the required functions to swap bytes. Use that and kick out <arpa/inet.h>